### PR TITLE
feat(client、permisson):client add and update，incidental fix permission relation bug

### PR DESCRIPTION
**Pull request by MayueCif**

### CODE REVIEW

File: src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Dtos/CheckItemDto.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿namespace Masa.Auth.Contracts.Admin.Infrastructure.Dtos;
+
+public class CheckItemDto<T> : IEquatable<CheckItemDto<T>>
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Dtos/CheckItemDto.cs:3]: T -> TId. T need clearer name

File: src/Contracts/Masa.Auth.Contracts.Admin/Sso/ClientDetailDto.cs [hunk 2]
```diff
@@ -1,5 +1,86 @@
 ﻿namespace Masa.Auth.Contracts.Admin.Sso;
 
-public class ClientDetailDto
+public class ClientDetailDto : ClientAddDto
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Sso/ClientDetailDto.cs:3]: ClientAdd? AddClient?

File: src/Services/Masa.Auth.Service.Admin/Application/Sso/CommandHandler.cs [hunk 3]
```diff
@@ -2,15 +2,74 @@
...
+        {
+            switch (client.ClientType)
+            {
+                case ClientTypes.Web:
+                    client.AllowedGrantTypes.AddRange(GrantTypeConsts.Code);
+                    client.RequirePkce = true;
+                    client.RequireClientSecret = true;
+                    break;
+                case ClientTypes.Spa:
+                    client.AllowedGrantTypes.AddRange(GrantTypeConsts.Code);
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Sso/CommandHandler.cs:33]: Spa is same with Native. You can remove 33 to 36 lines.

File: src/Services/Masa.Auth.Service.Admin/Application/Sso/QueryHandler.cs [hunk 4]
```diff
@@ -2,31 +2,59 @@
...
+    [EventHandler]
+    public void ClientTypeListAsync(ClientTypeListQuery clientTypeListQuery)
+    {
+        clientTypeListQuery.Result = Enum<ClientTypes>.GetItems()
+            .Select(ct => new ClientTypeDetailDto
+            {
+                ClientType = ct,
+                Description = ct switch
+                {
+                    ClientTypes.Web => "Server-side applications where authentication and tokens are handled on the server (for example, Go, Java, ASP.Net, Node.js, PHP)",
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Sso/QueryHandler.cs:41]: Add a todo here. I18n will be supported in future.

File: src/Services/Masa.Auth.Service.Admin/Domain/Sso/Aggregates/Client.cs [hunk 5]
```diff
@@ -93,6 +95,9 @@ public class Client : AuditAggregateRoot<int, Guid>, ISoftDelete
     public List<ClientCorsOrigin> AllowedCorsOrigins { get; private set; } = new();
 
     public List<ClientProperty> Properties { get; private set; } = new();
+    //public IReadOnlyCollection<ClientProperty> Properties => properties;
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Domain/Sso/Aggregates/Client.cs:98]: Remove the comment.

File: src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs [hunk 6]
```diff
@@ -15,10 +15,24 @@ public void Configure(EntityTypeBuilder<Permission> builder)
             v => v.ToString(),
             v => (PermissionTypes)Enum.Parse(typeof(PermissionTypes), v)
         );
-        builder.HasMany(p => p.Permissions).WithOne(pi => pi.ChildPermission).HasForeignKey(pi => pi.ChildPermissionId);
         builder.HasMany(p => p.RolePermissions).WithOne(rp => rp.Permission);
         builder.HasMany(p => p.UserPermissions).WithOne(up => up.Permission);
         builder.HasMany(p => p.TeamPermissions).WithOne(tp => tp.Permission);
+        builder.HasMany(p => p.ParentPermissions).WithMany(pi => pi.ChildPermissions)
+            .UsingEntity<PermissionRelation>(
+                    j => j
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs:23]: Why j?

File: src/Web/Masa.Auth.Web.Admin.Rcl/wwwroot/i18n/zh-CN.json [hunk 7]
```diff
@@ -199,5 +199,8 @@
   "IdentityResource": "身份",
   "IdentityResource.Name": "配置名称",
   "IdentityResource.Required": "是否必须",
-  "User Claims": "用过声明"
+  "User Claims": "用过声明",
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Admin.Rcl/wwwroot/i18n/zh-CN.json:202]: 用过？

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs [hunk 8]
```diff
@@ -20,18 +20,18 @@ public void Configure(EntityTypeBuilder<Permission> builder)
         builder.HasMany(p => p.TeamPermissions).WithOne(tp => tp.Permission);
         builder.HasMany(p => p.ParentPermissions).WithMany(pi => pi.ChildPermissions)
             .UsingEntity<PermissionRelation>(
-                    j => j
+                    cr => cr
                     .HasOne(pr => pr.ParentPermission)
                     .WithMany(p => p.ChildPermissionRelations)
                     .HasForeignKey(pr => pr.ParentPermissionId),
-                    j => j
+                    cl => cl
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs:27]: Why cl?

File: src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs [hunk 9]
```diff
@@ -20,18 +20,18 @@ public void Configure(EntityTypeBuilder<Permission> builder)
...
                     .HasOne(pr => pr.ChildPermission)
                     .WithMany(p => p.ParentPermissionRelations)
                     .HasForeignKey(pr => pr.ChildPermissionId),
-                    j =>
+                    cj =>
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs:31]: Why cj?

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Sso/Client/UpdateClientDialog.razor.cs [hunk 10]
```diff
@@ -22,20 +24,24 @@ public partial class UpdateClientDialog
     ClientAuthenticationDto _authenticationDto = new();
     ClientDeviceFlowDto _deviceFlowDto = new();
     ClientTokenDto _tokenDto = new();
+    ClientCredentialDto _clientCredentialDto = new();
+    Type _otherType = null!;
+
+    Dictionary<string, ComponentMetadata> _componentParameters = new();
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Sso/Client/UpdateClientDialog.razor.cs]: Why _componentParameters have a blank line above.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs (hunk 8 above)
Comment (MayueCif) [reply src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs:27]: configleft

File: src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs (hunk 9 above)
Comment (MayueCif) [reply src/Services/Masa.Auth.Service.Admin/Infrastructure/EntityConfigurations/Permissions/PermissionEntityTypeConfiguration.cs:31]: configjoin

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Sso/Client/UpdateClientDialog.razor.cs (hunk 10 above)
Comment (MayueCif) [reply src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Sso/Client/UpdateClientDialog.razor.cs]: Dictionary<string, ComponentMetadata> is already to Dictionary<string, object> and delete blank line

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**